plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/button/block.tsx

```diff
@@ -59,6 +63,31 @@ const getButtonText = ( {
 	return productCartDetails?.text || __( 'Add to cart', 'woocommerce' );
 };
 
+const AddToCartButtonAdminSide = ( {
+	product,
+}: {
+	product: ProductEntityResponse;
+} ): JSX.Element => {
+	const isExternal = product.type === 'external';
+	return (
+		<button
+			disabled={ false }
+			className={ clsx(
+				'wp-block-button__link',
+				'wp-element-button',
+				'add_to_cart_button',
+				'wc-block-components-product-button__button'
+			) }
+			style={ {} }
+		>
+			{ isExternal
+				? product.button_text
+				: product.add_to_cart?.single_text ||
```

Comment:
Ideally, we should use `isDescendantOfAddToCartWithOptions` to decide whether we need to render `single_text` (when rendered inside the _Add to Cart + Options_ block) or `text` (when rendered anywhere else).

The issue can be seen with this snippet:

```PHP
add_filter( 'woocommerce_product_add_to_cart_text', function() {
	return 'Button text in the shop page';
});
add_filter( 'woocommerce_product_single_add_to_cart_text', function() {
	return 'Button text in the product page';
});
```

If you edit the Shop page, you will notice the `'Button text in the product page'` string is used in the editor, instead of `'Button text in the shop page'`.

<img width="2147" height="981" alt="imatge" src="https://github.com/user-attachments/assets/9ea81e1b-58fa-49dc-b9e7-3147420da448" />

Reply:
Good point! Fixed with 1fa4cb9d887ca6aec4070694b833aae89d0c123d. For the external product, we still need to use the `.button_text` property.